org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java

```diff
@@ -66,8 +74,23 @@ public void afterEach(ExtensionContext context) throws Exception {
 	}
 
 
+	private static InjectWith getInjectWith(Class<?> cls) {
```

Comment:
@kthoms @szarnekow this is the algorithm i proposed in the bugzilla.
what do you think?

Reply:
Looks good to me.